types/hoist-non-react-statics/index.d.ts

```diff
@@ -47,4 +47,4 @@ declare function hoistNonReactStatics<
         >]: S[key]
     };
 
-export = hoistNonReactStatics;
+export default hoistNonReactStatics;
```

Comment:
It seems like the [underlying library](https://unpkg.com/hoist-non-react-statics@3.0.1/dist/hoist-non-react-statics.cjs.js) does not use `export default`, so the .d.ts file should not use it either. If you would like to use `import hoistNonReactStatics from 'hoist-non-react-statics';` in consuming code, you can use the `esModuleInterop` compilation flag.

Reply:
@uniqueiniquity Uses export default here: https://github.com/mridgway/hoist-non-react-statics/blob/master/src/index.js#L43
```
import hoistNonReactStatics from 'hoist-non-react-statics'
``` 
does not work with `esModuleInterop` turned on:
```
 - error TS1192: Module '"...hoist-non-react-statics/index"' has no default export.

2 import hoistNonReactStatics from "hoist-non-react-statics";
```
It does however work with this patch.

Reply:
Aha, I see. Unfortunately, it seems like that library is using rollup to compile away that `export default`. You can see at https://runkit.com/embed/f1ioq85royf2 that the default property on the module doesn't actually exist. This all works fine in TS with your proposed changes when the `esModuleInterop` flag is enabled, but since we can't guarantee that users have that flag enabled (or even available, since it's relatively new), it's not a safe change to make (i.e. running the following with `esModuleInterop` off gives a runtime error).
```
import hoistNonReactStatics from 'hoist-non-react-statics';
hoistNonReactStatics({} as React.ComponentType<any>, {} as React.ComponentType<any>);
```
As far as the compiler errors you're hitting, do you have the `allowSyntheticDefaultImports` flag explicitly set to false? That would cause this error.

Reply:
Yeah, had to set `allowSyntheticDefaultImports` to true. Thanks for the help!